### ISSUE TITLE
Seemingly incorrect errors, potentially related to C++11

### ISSUE THREAD

**Issue by emilydolson** [hebaishi/linter-gcc#58]
## Checklist
* [x] Check that you have the GNU C Compiler installed by typing ```gcc --version``` into a terminal.
* [x] If you don't have the ```linter``` package installed, either update to the latest version of ```linter-gcc``` or install linter directly with ```apm install linter```.
* [x] Ensure that you are using a valid grammar - either C/C++/C++14

## Issue (including debugging messages):
When I compile my program with g++ myccfile.cc -std=c++11 -Wall, it works fine (all I get are a few warnings about unused functions). The [clang linter](https://atom.io/packages/linter-clang) shows the same warnings, but nothing else. 

linter-gcc, however, shows me a ton of errors. In the .cc file I'm compiling, I'm getting a "No such file or directory" error on my first include (regardless of whether I add it to the include path). It's definitely the right file path because it compiles fine. The first error for the project as a whole is 

> "#error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.at line 32 col 2 in /usr/include/c++/4.8/bits/c++0x_warning.h".

despite the fact that I have -std=c++11 in my flags. There are also a number of warnings about the fact that I'm using C++11 reserved words (I'm using them for their intended purpose). I'm using the C++14 grammar (using the C++ grammar doesn't fix it).

Here are the debugging messages I get on the console when linter-gcc runs:

> linter-gcc: Using configuration page settings
> /home/emily/.atom/packages/linter-gcc/lib/utility.js:94 linter-gcc config: {"execPath":"/usr/bin/g++","gccIncludePaths":"./evo/*, ./tools/*","gccSuppressWarnings":false,"gccDefaultCFlags":"-std=c++11","gccDefaultCppFlags":"-c -std=c++11","gccErrorLimit":0,"gccErrorString":"error","gccWarningString":"warning","gccNoteString":"note"}
> 2/home/emily/.atom/packages/linter-gcc/lib/utility.js:76 Expanding directories
> /home/emily/.atom/packages/linter-gcc/lib/utility.js:146 linter-gcc: /usr/bin/g++ -c -std=c++11 -fmax-errors=0 -I/home/emily/repos/Empirical/evo -I/home/emily/repos/Empirical/tools /tmp/5b9837c7-122b-4820-8092-d54549d3bf68.cpp

More information in case this is hard to duplicate:
- My gcc version is 4.8.4 (I've also tried it with 4.9)
- I'm running Linux Mint 17 64-bit
- The project I'm having this problem with is [here](https://github.com/emilydolson/Empirical). It's largely a pre-processor library, so it's possible that all of the (somewhat convoluted) macros and templates are confusing the linter. [This](https://github.com/emilydolson/Empirical/blob/master/examples/evo/NK_OEE.cc) is the specific test file I'm trying to compile.

Thanks so much! This otherwise seems like the best of the atom C++ linters.

**Comment by hebaishi** [comment]
I've cloned the repo and it works fine for me. I think it's just a path issue. This is my linter-gcc config:
```json
{
    "execPath": "g++",
    "gccIncludePaths": "./include",
    "gccSuppressWarnings": false,
    "gccDefaultCFlags": "-c -Wall",
    "gccDefaultCppFlags": "-c -Wall -Wextra -Wpedantic -std=c++14 -o /dev/null",
    "gccErrorLimit": -1,
    "gccErrorString": "error",
    "gccWarningString": "warning",
    "gccNoteString": "note"
}
```

Could you please try using the same config and seeing if it works. The most important option here is `gccIncludePaths`.

**Comment by emilydolson** [comment]
I tried using that config, but I'm still getting the same errors. I updated Atom to the latest version and the number of issues went down from ~250 to 56. Adding "./*" to the include path made the "File not found" error in my .cc file go away, but there are still some other seemingly erroneous "File not found" errors (for instance "tools/vector.h" is never found).

**Comment by hebaishi** [comment]
Sorry for the late response.. I've been a bit busy. If you're still having the problem of missing files, could you do the following:
1. Enable linter-gcc's debugging messages in the configuration page
2. Open the developer console by pressing `Ctrl+Shift+I`
3. Lint a C/C++ file by saving it/using lint on-the-fly.
4. Report the linter-gcc command that is printed to the console. Also paste the command into the terminal to check the output,

**Comment by hebaishi** [comment]
Hello. Have you managed to get the problem fixed? If not, could you please follow the instructions above and report the linter-gcc command?

**Comment by hebaishi** [comment]
Hello. Any updates on the errors?

**Comment by emilydolson** [comment]
Sorry for the delay in getting back to you - I've been traveling. It looks like it's working now. The update must have fixed it. Thanks!

**Comment by higorhi** [comment]
I now have a similar problem. I have two files, List.h and test.cpp, in the same folder. The linter is reporting, for the test.cpp file, the error "List.h: No such file or directory", though it compiles just fine with g++ through command lines and through gpp-compiler in Atom.

I'm not a very experienced programmer, but I really like Atom and would appreciate to continue using it, but I see these errors might be getting in my way, because they prevent the package from linting the rest of my code, for some reason.

I'd appreciate some help! Thanks.

**Comment by higorhi** [comment]
It seems that disabling linting on-the-fly solves the problem, though I don't have any auto-saving packages (I uninstalled gpp-compiler), as is warned in the setting's description.

Thanks for the help, again.

**Comment by eencae** [comment]
Adding -I (uppercase i, not lowercase L) to the compile flags tells the compiler where to find the header files and solves the "No such file or directory" problem.
